base_wamas_ubl/lib/wamas/utils.py

```diff
@@ -275,6 +319,9 @@ def generate_wamas_line(dict_item, grammar, **kwargs):  # noqa: C901
                 args = (dict_wamas_out,)
                 args += ast.literal_eval(re.search(r"\((.*?)\)", df_func).group(0))
                 df_func = "get_date_from_field"
+            elif "get_Adrs_" in df_func:
```

Comment:
TBH this way of passing arguments seems a bit weird, fragile and hard to understand overall.
The `df_func` or another key in the handlers mapping should provide a way to determine those args.
You could make them all classes instead of simple methods and accept parameters and get back arguments...
Do not change it now but can you please add a TODO to cleanup this?